tests/gateway/test_utils.py

```diff
@@ -0,0 +1,139 @@
+    get_gateway_uri,
+    SearchRoutesToken,
+)
+import mlflow.gateway.envs
+
+
+@pytest.mark.parametrize(
+    "base_url, route",
+    [
+        ("http://127.0.0.1:6000", "gateway/test/invocations"),
+        ("http://127.0.0.1:6000/", "/gateway/test/invocations"),
+        ("http://127.0.0.1:6000/gateway", "/test/invocations"),
+        ("http://127.0.0.1:6000/gateway/", "/test/invocations"),
+        ("http://127.0.0.1:6000", "gateway/test/invocations"),
+        ("http://127.0.0.1:6000/", "/gateway/test/invocations"),
+        ("http://127.0.0.1:6000/gateway", "test/invocations"),
+        ("http://127.0.0.1:6000/gateway/", "test/invocations"),
+    ],
+)
+def test_resolve_route_url(base_url, route):
+    assert resolve_route_url(base_url, route) == "http://127.0.0.1:6000/gateway/test/invocations"
+
+
+@pytest.mark.parametrize("base_url", ["databricks", "databricks://my.workspace"])
+def test_resolve_route_url_qualified_url_ignores_base(base_url):
+    route = "https://my.databricks.workspace/api/2.0/gateway/chat/invocations"
+
+    resolved = resolve_route_url(base_url, route)
+
+    assert resolved == route
+
+
+@pytest.mark.parametrize(
+    "name, expected",
+    [
+        ("validName", True),
+        ("invalid name", False),
+        ("invalid/name", False),
+        ("invalid?name", False),
+        ("", False),
+    ],
+)
+def test_is_valid_endpoint_name(name, expected):
+    assert is_valid_endpoint_name(name) == expected
+
+
+def test_check_configuration_route_name_collisions():
+    config = {"routes": [{"name": "name1"}, {"name": "name2"}, {"name": "name1"}]}
+    with pytest.raises(MlflowException, match="Duplicate names found in route configurations"):
+        check_configuration_route_name_collisions(config)
+
+
+@pytest.mark.parametrize(
+    "uri, expected",
+    [
+        ("http://localhost", True),
+        ("databricks", True),
+        ("localhost", False),
+        ("http:/localhost", False),
+        ("", False),
+    ],
+)
+def test__is_valid_uri(uri, expected):
+    assert _is_valid_uri(uri) == expected
+
+
+@pytest.mark.parametrize(
+    "paths, expected",
+    [
+        (["path1", "path2", "path3"], "/path1/path2/path3"),
+        (["/path1/", "/path2/", "/path3/"], "/path1/path2/path3"),
+        (["/path1//", "/path2//", "/path3//"], "/path1/path2/path3"),
+        (["path1", "", "path3"], "/path1/path3"),
+        (["", "", ""], "/"),
+        ([], "/"),
+    ],
+)
+def test_assemble_uri_path(paths, expected):
+    assert assemble_uri_path(paths) == expected
+
+
+def test_set_gateway_uri(monkeypatch):
+    monkeypatch.setattr("mlflow.gateway.utils._gateway_uri", None)
+
+    valid_uri = "http://localhost"
+    set_gateway_uri(valid_uri)
+    assert mlflow.gateway.utils._gateway_uri == valid_uri
+
+    invalid_uri = "localhost"
+    with pytest.raises(MlflowException, match="The gateway uri provided is missing required"):
+        set_gateway_uri(invalid_uri)
+
+
+def test_get_gateway_uri(monkeypatch):
```

Comment:
`get_gateway_uri` is not tested in `tests/gateway/test_fluent.py`?

Reply:
It is :) I like the idea of having each of these utils tested within this test suite, though. Just to provide a direct indicator of util logic issues without any additional test logic apart from the exclusive validation of the utility function